src/midi/midi_fluidsynth.cpp

```diff
@@ -44,6 +42,149 @@
+
+int  (*fluid_synth_chorus_on)(fluid_synth_t *synth, int fx_group, int on) = nullptr;
+int  (*fluid_synth_set_chorus_group_nr)(fluid_synth_t *synth, int fx_group, int nr) = nullptr;
+int  (*fluid_synth_set_chorus_group_level)(fluid_synth_t *synth, int fx_group, double level) = nullptr;
+int  (*fluid_synth_set_chorus_group_speed)(fluid_synth_t *synth, int fx_group, double speed) = nullptr;
+int  (*fluid_synth_set_chorus_group_depth)(fluid_synth_t *synth, int fx_group, double depth_ms) = nullptr;
+int  (*fluid_synth_set_chorus_group_type)(fluid_synth_t *synth, int fx_group, int type) = nullptr;
+
+int  (*fluid_synth_reverb_on)(fluid_synth_t *synth, int fx_group, int on) = nullptr;
+int  (*fluid_synth_set_reverb_group_roomsize)(fluid_synth_t *synth, int fx_group, double roomsize) = nullptr;
+int  (*fluid_synth_set_reverb_group_damp)(fluid_synth_t *synth, int fx_group, double damping) = nullptr;
+int  (*fluid_synth_set_reverb_group_width)(fluid_synth_t *synth, int fx_group, double width) = nullptr;
+int  (*fluid_synth_set_reverb_group_level)(fluid_synth_t *synth, int fx_group, double level) = nullptr;
+
+int  (*fluid_synth_sfcount)(fluid_synth_t *synth) = nullptr;
+int  (*fluid_synth_sfload)(fluid_synth_t *synth, const char *filename, int reset_presets) = nullptr;
+void (*fluid_synth_set_gain)(fluid_synth_t *synth, float gain) = nullptr;
+int  (*fluid_synth_set_interp_method)(fluid_synth_t *synth, int chan, int interp_method) = nullptr;
+int  (*fluid_synth_noteoff)(fluid_synth_t *synth, int chan, int key) = nullptr;
+int  (*fluid_synth_noteon)(fluid_synth_t *synth, int chan, int key, int vel) = nullptr;
+int  (*fluid_synth_key_pressure)(fluid_synth_t *synth, int chan, int key, int val) = nullptr;
+int  (*fluid_synth_cc)(fluid_synth_t *synth, int chan, int ctrl, int val) = nullptr;
+int  (*fluid_synth_program_change)(fluid_synth_t *synth, int chan, int program) = nullptr;
+int  (*fluid_synth_channel_pressure)(fluid_synth_t *synth, int chan, int val) = nullptr;
+int  (*fluid_synth_pitch_bend)(fluid_synth_t *synth, int chan, int val) = nullptr;
+int  (*fluid_synth_sysex)(fluid_synth_t *synth, const char *data, int len,
+							char *response, int *response_len, int *handled, int dryrun) = nullptr;
+int  (*fluid_synth_write_float)(fluid_synth_t *synth, int len,
+								void *lout, int loff, int lincr,
+								void *rout, int roff, int rincr) = nullptr;
+
+/**
+ * A filthy macro to make resolving library symbols a little less verbose, 
+ * while still including robust error checking.
+ */
+#define DB_GET_FLUID_SYM(sym) sym = (decltype(sym))dynlib_get_symbol(fluid_lib, #sym); \
+	if (!sym) { dynlib_close(fluid_lib); return DynLibResult::ResolveSymErr; }
+
+/**
+ * Load the fluidsynth library and resolve all required symbols.
+ * 
+ * If the library is already loaded, does nothing.
+ */
+static DynLibResult load_dynlib_fluid()
+{
+	if (!fluid_lib) {
+		fluid_lib = dynlib_open("libfluidsynth-3" DB_FLUID_LIB_EXT);
```

Comment:
The name of the library file on my system is `libfluidsynth.so.3` which I think is going to be the most common on Linux. This is also the name I get when building fluidsynth from source. It's actually a symlink to `libfluidsynth.so.3.1.4` but the minor version will differ between distros.

After changing this and fixing the segfault issue, everything seems to work fine.

Reply:
I checked Debian and Fedora, and they're both using `libfluidsynth.so.3`, so switched to that.